includes/Export.php

```diff
@@ -559,83 +731,103 @@ function writeRevision( $row ) {
 		}
 
 		if ( $row->log_deleted & LogPage::DELETED_COMMENT ) {
-			$out .= "      " . Xml::element( 'comment', array( 'deleted' => 'deleted' ) ) . "\n";
+			$out .= "    " . Xml::element( 'comment', array( 'deleted' => 'deleted' ) ) . "\n";
 		} elseif ( $row->log_comment != '' ) {
-			$out .= "      " . Xml::elementClean( 'comment', null, strval( $row->log_comment ) ) . "\n";
+			$out .= "    " . Xml::elementClean( 'comment', null, strval( $row->log_comment ) ) . "\n";
 		}
 
-		$out .= "      " . Xml::element( 'type', null, strval( $row->log_type ) ) . "\n";
-		$out .= "      " . Xml::element( 'action', null, strval( $row->log_action ) ) . "\n";
+		$out .= "    " . Xml::element( 'type', null, strval( $row->log_type ) ) . "\n";
+		$out .= "    " . Xml::element( 'action', null, strval( $row->log_action ) ) . "\n";
 
 		if ( $row->log_deleted & LogPage::DELETED_ACTION ) {
-			$out .= "      " . Xml::element( 'text', array( 'deleted' => 'deleted' ) ) . "\n";
+			$out .= "    " . Xml::element( 'text', array( 'deleted' => 'deleted' ) ) . "\n";
 		} else {
 			$title = Title::makeTitle( $row->log_namespace, $row->log_title );
-			$out .= "      " . Xml::elementClean( 'logtitle', null, self::canonicalTitle( $title ) ) . "\n";
-			$out .= "      " . Xml::elementClean( 'params',
+			$out .= "    " . Xml::elementClean( 'logtitle', null, self::canonicalTitle( $title ) ) . "\n";
+			$out .= "    " . Xml::elementClean( 'params',
 				array( 'xml:space' => 'preserve' ),
 				strval( $row->log_params ) ) . "\n";
 		}
 
-		$out .= "    </logitem>\n";
+		$out .= "  </logitem>\n";
 
-		wfProfileOut( __METHOD__ );
 		return $out;
 	}
 
-	function writeTimestamp( $timestamp ) {
+	/**
+	 * @param string $timestamp
+	 * @param string $indent Default to six spaces
+	 * @return string
+	 */
+	function writeTimestamp( $timestamp, $indent = "      " ) {
 		$ts = wfTimestamp( TS_ISO_8601, $timestamp );
-		return "      " . Xml::element( 'timestamp', null, $ts ) . "\n";
+		return $indent . Xml::element( 'timestamp', null, $ts ) . "\n";
 	}
 
-	function writeContributor( $id, $text ) {
-		$out = "      <contributor>\n";
+	/**
+	 * @param int $id
+	 * @param string $text
+	 * @param string $indent Default to six spaces
+	 * @return string
+	 */
+	function writeContributor( $id, $text, $indent = "      " ) {
+		$out = $indent . "<contributor>\n";
 		if ( $id || !IP::isValid( $text ) ) {
-			$out .= "        " . Xml::elementClean( 'username', null, strval( $text ) ) . "\n";
-			$out .= "        " . Xml::element( 'id', null, strval( $id ) ) . "\n";
+			$out .= $indent . "  " . Xml::elementClean( 'username', null, strval( $text ) ) . "\n";
+			$out .= $indent . "  " . Xml::element( 'id', null, strval( $id ) ) . "\n";
 		} else {
-			$out .= "        " . Xml::elementClean( 'ip', null, strval( $text ) ) . "\n";
+			$out .= $indent . "  " . Xml::elementClean( 'ip', null, strval( $text ) ) . "\n";
 		}
-		$out .= "      </contributor>\n";
+		$out .= $indent . "</contributor>\n";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$out". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)